plan/task_profile.go

```diff
@@ -85,29 +85,39 @@ func attachPlan2TaskProfile(p PhysicalPlan, t taskProfile) taskProfile {
 // finishIndexPlan means we no longer add plan to index plan, and compute the network cost for it.
 func (t *copTaskProfile) finishIndexPlan() {
 	if t.addPlan2Index {
-		t.cst += float64(t.cnt) * netWorkFactor
+		t.indexPlan.SetSchema(expression.NewSchema()) // we only need the handle
```

Comment:
If the index scan is single read or has selection pushed down, we should not set it to empty.